DEPRECATION.md

```diff
@@ -0,0 +1,81 @@
+The `0.x` release series of the gRPC Swift project will be deprecated in favour of a new implementation.
+
+In the coming weeks this branch (currently `master`) containing the `0.x` releases will be renamed `cgrpc`. The `nio` branch containing the [new implementation][nio-branch] will subsequently be renamed to `master` and become the default branch. A `1.0.0` tag will also be created.
+
+Version | Branch Now | Branch After Deprecation
+--------|------------|-------------------------
+`0.x`   | `master`   | `cgrpc`
+`1.x`   | `nio`      | `master`
+
+## What is it being replaced with?
+
+We have rewritten gRPC Swift on top of [SwiftNIO][nio], an open-source asynchronous event-driven networking framework created by Apple. Our implementation will be written in Swift and will not wrap the gRPC core C library.
+
+## Why is this happening?
+
+There are a number of reasons we have rewritten gRPC Swift:
+
+- The `0.x` releases are built on top of a C interface to the gRPC core library provided by the [gRPC project][grpc]. This led to a number of memory safety issues and is easy to hold incorrectly.
+- Network connectivity changes (e.g. LTE to WiFi, 3G to LTE, etc.) are not handled well by the networking stack in the core library. [Network.framework][network-framework] (where available, see below) alleviates this problem and has integration with [SwiftNIO][nio].
+- [SwiftNIO][nio] has gained a lot of traction in the Swift on Server community due to its performance. We believe we can leverage this to improve the performance of gRPC Swift.
+- Vendoring copies of the gRPC core library and BoringSSL is a source of frustration for developers and users.
+
+## What will happen to the `0.x` releases?
+
+We will continue to patch the `cgrpc` branch for security fixes and serious bugs only. There will be no feature development on the `cgrpc` branch and the version of the gRPC core library will not be updated (unless necessary for a security fixes).
+
+## When is this happening?
+
+We plan to deprecate `0.x` and tag `1.0.0` by the end of the year.
+
+## Which Swift versions will be supported for `1.x`?
+
+Swift 5.0 and later.
+
+## Which platforms will be supported for `1.x`?
+
+We have the same [platform support as SwiftNIO][nio-platforms]. That is:
+
+* macOS 10.12+, iOS 10+
+* macOS 10.14+, iOS 12+, or tvOS 12+ (with [Network.framework][network-framework] support via [NIO Transport Services][nio-ts])
+* Ubuntu 14.04+
+
+## Which package managers will be supported for `1.x`?
+
+We will provide support for Swift Package Manager as we believe that its integration with Xcode is the most convient way to manage packages.
+
+Carthage will _not_ be supported as it has been the source of a [number][carthage0] [of][carthage1] [issues][carthage2] [in][carthage3] [the][carthage4] [past][carthage5].
+
+## Can I try this out now?
+
+Absolutely! Head over to the `nio` [branch][nio-branch] and check out the [quick-start guide][quick-start] or [basic tutorial][basic-tutorial].
```

Comment:
Also mention (maybe further up) that we consider `nio` production-ready already?